File: extract_logo.py

import base64
import re

html_path = r'C:\Users\luan.muniz\Documents\RVGS ELETRICA\RVGS ELETRICA\html\rvgs-eletrica (4).html'
output_path = r'c:\Users\luan.muniz\Documents\RVGS ELETRICA\RVGS ELETRICA\public\logo_transparent.png'

with open(html_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Find the first data:image/png;base64 string
match = re.search(r'src="data:image/png;base64,([^"]+)"', content)
if match:
    base64_data = match.group(1)
    img_data = base64.b64decode(base64_data)
    with open(output_path, 'wb') as f:
        f.write(img_data)
    print(f"Logo saved successfully to {output_path}")
else:
    print("Could not find base64 logo in HTML")

File: extract_logo.js

const fs = require('fs');
const path = require('path');

const htmlPath = 'C:\\Users\\luan.muniz\\Documents\\RVGS ELETRICA\\RVGS ELETRICA\\html\\rvgs-eletrica (4).html';
const outputPath = 'C:\\Users\\luan.muniz\\Documents\\RVGS ELETRICA\\RVGS ELETRICA\\public\\logo_transparent.png';

const content = fs.readFileSync(htmlPath, 'utf8');
const match = content.match(/src="data:image\/png;base64,([^"]+)"/);

if (match) {
    const base64Data = match[1];
    const buffer = Buffer.from(base64Data, 'base64');
    fs.writeFileSync(outputPath, buffer);
    console.log(`Logo saved successfully to ${outputPath}`);
} else {
    console.log("Could not find base64 logo in HTML");
}
